pkg/common/peertracker/getpid.c

```diff
@@ -0,0 +1,52 @@
+#include <windows.h>
+#include <iphlpapi.h>
+#include <stdio.h>
+
+unsigned long getOwningPIDFromLocalConn(int localPort, int remotePort, int *pid)
+{
+	PMIB_TCPTABLE_OWNER_PID pTcpTable = NULL;
+	DWORD dwSize = 0;
+	unsigned long retCode = 0;
+
+	for (;;) {
+		if (dwSize > 0) {
+			pTcpTable = (PMIB_TCPTABLE_OWNER_PID)calloc(dwSize, 1);

```

Comment:
This will leak pTcpTable if we had to reallocate because the table grew after the second call. We should either always free pTcpTable at the end of the loop or use `realloc` here to extend the existing allocation.

Reply:
Thanks for catching that. Just calling `free(pTcpTable)` at the end of the loop seems to be more simple.